vars/azureBlobUpload.groovy

```diff
@@ -1,12 +1,13 @@
-def call(String credentialsId, String source, String destination) {
-  withDocker('hmcts/moj-azcopy-image:7.2.0-netcore-1.0', '-u root') {
+def call(String subscription, String credentialsId, String source, String destination) {
     withCredentials([usernamePassword(credentialsId: credentialsId, passwordVariable: 'STORAGE_ACCOUNT_KEY', usernameVariable: 'STORAGE_ACCOUNT_NAME')]) {
-      withEnv(["SOURCE=${source}", "DESTINATION=${destination}"]) {
-        sh 'azcopy --source ${SOURCE} \
-          --destination https://${STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${DESTINATION} \
-          --dest-key ${STORAGE_ACCOUNT_KEY} \
-          --set-content-type --recursive'
+      withSubscriptionLogin(subscription) {
+        withEnv(["SOURCE=${source}", "DESTINATION=${destination}"]) {
+          sh 'sudo chmod +x /usr/bin/azcopy && \
```

Comment:
we should not be using sudo in our pipelines, if its permissions are wrong it should be fixed in the packer image

Reply:
Agent has been updated and sudo removed. Tested and working